SeeSharp.Templates/content/SeeSharp.Blazor.Template/Pages/Experiment.razor.cs

```diff
@@ -1,60 +1,319 @@
 using Microsoft.AspNetCore.Components;
+using Microsoft.JSInterop;
 using SeeSharp.Blazor;
 
 namespace SeeSharp.Blazor.Template.Pages;
 
-public partial class Experiment : ComponentBase
-{
+// Only here for the example to show some possibilities
+struct ExampleImageGenerator {
+    public Image rndImage(float strength, int width, int height, bool colored) {
+        Image image = new Image(width, height, 3);
+        RNG rng = new RNG();
+
+        for (int y = 0; y < height; y++) {
+            for (int x = 0; x < width; x++) {
+                if (colored) {
+                    image.SetPixelChannel(x, y, 0, rng.NextFloat(0.5f - strength, 0.5f + strength));
+                    image.SetPixelChannel(x, y, 1, rng.NextFloat(0.5f - strength, 0.5f + strength));
+                    image.SetPixelChannel(x, y, 2, rng.NextFloat(0.5f - strength, 0.5f + strength));
+                } else {
+                    float value = rng.NextFloat(0.5f - strength, 0.5f + strength);
+
+                    image.SetPixelChannel(x, y, 0, value);
+                    image.SetPixelChannel(x, y, 1, value);
+                    image.SetPixelChannel(x, y, 2, value);
+                }
+
+            }
+        }
+
+        return image;
+    }
+}
+
+struct ListenerState {
+    public ListenerState() { }
+
+    /// <summary>
+    /// The index of the selected image of the current selected flipbook (selected by clicking on it)
+    /// </summary>
+    public int selectedIndex = 0;
+
+    /// <summary>
+    /// Number between 0 and NumSamples. Can be used if data is stored from different iterations
+    /// </summary>
+    public int currIteration = 0;
+
+    public bool altKeyPressed = false;
+    public bool ctrlKeyPressed = false;
+    public int currX = 0;
+    public int currY = 0;
+
+    /// <summary>
+    /// The key of the current flipbook in string form and concatenated with ','
+    /// </summary>
+    public string currFlipKey = "";
+}
+
+/// <summary>
+/// Differences between event type so update methods for flipbooks can ignore events
+/// </summary>
+enum FiredType {
+    Click = 0,
+    Move = 1,
+    Wheel = 2,
+    KeyDown = 4,
+    KeyUp = 8,
+}
```

Comment:
looks like this is generic enough to become part of the SeeSharp.Blazor lib rather than cluttering the experiment code